### PR TITLE
Add quick and dirty beckmark

### PR DESCRIPTION
**Pull request by jcarres-mdsol**
@ejennings-mdsol @cmcinnes-mdsol 
Quick benchmark for the library, I will not be available for a couple of weeks so ok if you want to close this in favor of something more decent.

Run it with `rake benchmark`

### CODE REVIEW

**Comment by ejennings-mdsol** [comment]
Thanks @jcarres-mdsol!

I had to add `spec.add_development_dependency 'benchmark-ips', '~> 2.7'` to `mauth-client.gemspec` to get it to work.

**Comment by jcarres-mdsol** [comment]
@ejennings-mdsol  Thanks. I guess I had that gem installed globally? 
Anyways, added it to the develpment dependencies.

File: Rakefile [hunk 1]
```diff
@@ -1,6 +1,72 @@
 require 'bundler/gem_tasks'
 require 'rspec/core/rake_task'
+require 'mauth/request_and_response'
+require 'mauth/client'
+require 'securerandom'
+require 'benchmark/ips'
+require 'faraday'
+require "rspec/mocks/standalone"
```
Comment (jfeltesse-mdsol) [inline Rakefile]: throughout the file you're using double and single quotes, pick one please

File: Rakefile [hunk 2]
```diff
@@ -1,6 +1,72 @@
...
+class TestSignableRequest < MAuth::Request
+  include MAuth::Signed
+  attr_accessor :headers
+  def merge_headers(headers)
+    self.class.new(@attributes_for_signing).tap{|r| r.headers = (@headers || {}).merge(headers) }
+  end
+  def x_mws_time
+    headers['X-MWS-Time']
+  end
+  def x_mws_authentication
```
Comment (jfeltesse-mdsol) [inline Rakefile]: [nits] have empty lines between methods for readability? I understand it's utility class you might want to keep compact but I think the pros overweight the cons as far as new lines are concerned

File: Rakefile [hunk 3]
```diff
@@ -1,6 +1,72 @@
...
+    headers['X-MWS-Authentication']
+  end
+end
+
+desc 'Runs benchmarks for the library.'
+task :benchmark do
+  mc = MAuth::Client.new(private_key: OpenSSL::PKey::RSA.generate(2048), app_uuid: SecureRandom.uuid)
+  authenticating_mc = MAuth::Client.new(:mauth_baseurl => 'http://whatever', :mauth_api_version => 'v1')
```
Comment (jfeltesse-mdsol) [inline Rakefile]: use the new hash syntax?

**Comment by ejinotti-mdsol** [comment]
Thanks, LGTM 👍

**Pull request merged**